cdap-docs/deploy.sh

```diff
@@ -0,0 +1,165 @@
+  TMP_VERSION=`grep "<version>" ../pom.xml`
+  TMP_VERSION=${TMP_VERSION#*<version>}
+  VERSION=${TMP_VERSION%%</version>*}
+}
+
+function get_project () {
+  TMP_PROJECT=`grep "<artifactId>" ../pom.xml`
+  TMP_PROJECT=${TMP_PROJECT#*<artifactId>}
+  PROJECT=${TMP_PROJECT%%</artifactId>*}
+}
+
+function convert_branch_name () {
+  decho "converting '/' to '-' in branch name"
+  DOC_DIR=`echo $DOC_DIR | tr '/' '-'`
+}
+
+set_remote_dir () {
+  DOC_DIR=${OPT_DIR:-${BRANCH_NAME}}
+  convert_branch_name
+  if [[ "${DOC_DIR}" == release* || "${DOC_DIR}" == develop* ]]; then
+    DIR=''
+  else
+    DIR=branches/${DOC_DIR}
+  fi
+  decho "SUBDIR=${DIR}"
+}
+
+# parameters that can be passed to this script (as environment variables)
+DEBUG=${DEBUG:-no}
+DEPLOY_TO_STG=${DEPLOY_TO_STG:-no}
+DEPLOY_TO_DOCS=${DEPLOY_TO_DOCS:-no}
+BUILD_WORKING_DIR=${BUILD_WORKING_DIR:-/var/bamboo/xml-data/build-dir/CDAP-DRBD-JOB1}
+DOCS_SERVER1=${DOCS_SERVER1:-docs1.cask.co}
+DOCS_SERVER2=${DOCS_SERVER2:-docs2.cask.co}
+STG_SERVER=${STG_SERVER:-docs-staging.cask.co}
+REMOTE_STG_BASE=${REMOTE_STG_BASE:-/var/www/html/cdap}
+REMOTE_DOCS_BASE=${REMOTE_DOCS_BASE:-/var/www/docs/cdap}
+
+decho "OPT_DIR=${OPT_DIR}"
+decho "BRANCH_NAME=${BRANCH_NAME}"
+
+get_version
+get_project
+set_remote_dir
+
+#
+USER=bamboo
+PROJECT_DOCS=${PROJECT}-docs
+WEB_FILE=${PROJECT}-docs-${VERSION}-web.zip
+GITHUB_FILE=${PROJECT}-docs-${VERSION}-github.zip
+FILE_PATH=${BUILD_WORKING_DIR}/${PROJECT}/${PROJECT_DOCS}/build
+DOCS_SERVERS="${DOCS_SERVER1} ${DOCS_SERVER2}"
+REMOTE_STG_DIR="${REMOTE_STG_BASE}/${DIR}"
+REMOTE_DOCS_DIR="${REMOTE_DOCS_BASE}/${DIR}"
+
+RSYNC_OPTS='-aPh'
+SSH_OPTS='ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null'
+RSYNC_PATH='sudo rsync'
+
+die ( ) { echo ; echo "ERROR: ${*}" ; echo ; exit 1; }
+
+if [ "${PROJECT}" == '' ]; then
+  echo "PROJECT not defined"
+  exit 1
+fi
+
+################################################################################
+
+function make_remote_dir () {
+  decho "making sure remote directory ${3} exists on ${2}"
+  decho "ssh ${1}@${2} \"sudo mkdir -p ${3}\""
+  ssh ${1}@${2} "sudo mkdir -p ${3}" || die "could not create ${3} directory on ${2}"
+  decho ""
+}
+
+function rsync_zip_file () {
+  decho "rsyncing archive ${4} to ${2}"
+  decho "rsync ${RSYNC_OPTS} -e \"${SSH_OPTS}\" --rsync-path=\"${RSYNC_PATH}\" ${5}/${4} \"${1}@${2}:${3}/.\"" || die "could not rsync ${4} to ${2}"
+  rsync ${RSYNC_OPTS} -e "${SSH_OPTS}" --rsync-path="${RSYNC_PATH}" ${5}/${4} "${1}@${2}:${3}/." || die "could not rsync ${4} to ${2}"
+  decho ""
+}
+
+function unzip_archive () {
+  decho "unzipping ${4} on ${2}"
+  decho "ssh ${1}@${2} \"sudo unzip -o ${3}/${4} -d ${3}\""
+  ssh ${1}@${2} "sudo unzip -o ${3}/${4} -d ${3}" || die "unable to unzip ${4} in ${3} on ${2}, as ${1}"
+  decho ""
+}
+
+function deploy () {
+  decho "deploying to ${2}"
+  make_remote_dir ${1} ${2} ${3}
+  rsync_zip_file ${1} ${2} ${3} ${4} ${5}
+  unzip_archive ${1} ${2} ${3} ${4}
```

Comment:
With numbers instead names, it's hard for me to tell if this is working or going to work. But my first take is that it's not going to.

The existing staging script rsync the file, unzips into a temp directory, copies the zip into the unzipped archive -- which will be named VERSION-[SNAPSHOT], and then moves and renames that version to the version with the branch name. It doesn't look like you are doing this. 

It looks like the branches are becoming a sub-directory, and the version goes under that? Because currently all versions are in the same directory, and named by their branch. See http://stg-web101.sw.joyent.continuuity.net/cdap/
I would prefer to keep the current structure, as it's easy to find things.

Even with that, this looks like things might end up in the wrong location. Hard to tell without running it.

I don't see the step to copy the zip file inside the new directory, so it can be downloaded.